### ISSUE TITLE
pyGSTi Refactor Issue Thread

### ISSUE THREAD

**Issue by coreyostrove** [pyGSTio/pyGSTi#204]
Since many small issues are anticipated in light of the current refactor of the pyGSTi codebase this issue will serve as a thread for gathering and tracking them. Please feel free to continue opening up separate issue threads as you deem appropriate, especially for more major bugs or feature requests.

**Comment by coreyostrove** [comment]
###Unexpected behavior in LindbladErrorgen:###
I noticed a few instances of behavior using the LindbladErrorgen object that were unexpected. The first is related to the inclusion of basis labels for diagonal stochastic error rates in the lindblad_term_dict. The documentation and "Operators" tutorial indicate that a single basis label is sufficient for a diagonal term but post-refactor both are now needed. I don't know that anything needs to change here but the docstrings/tutorials should be amended to reflect this.

The second instance is more of a behavior change request. When specifying off-diagonal terms in the stochastic rates tensor using lindblad_term_dict I think that pyGSTi should automatically impose the symmetry (hermiticity) constraint when specifying only one half of a pair of off-diagonal terms and fill out the rest of the tensor for you. By default it doesn't and this raises an error. 

I have attached a jupyter notebook with examples of both behaviors. This was run on commit# b599f4f1 on the develop branch. 

[lindbladerrorgen_unexpected_behavior.zip](https://github.com/pyGSTio/pyGSTi/files/7042436/lindbladerrorgen_unexpected_behavior.zip)

**Comment by sserita** [comment]
Saw the following error while running the algorithms/ModelTesting Jupyter notebook. Probably just a kwarg mismatch.

```
/Users/sserita/Documents/repos/pyGSTi-tutorials/pygsti/report/factory.py:1242: UserWarning:

Idle tomography failed:
Can't cast <class 'pygsti.evotypes.evotype.Evotype'> to be a basis!
```

**Comment by sserita** [comment]
Unclear how to specify noise on the `"(auto_global_idle)"` gate constructed in `LocalNoiseModel` (either in constructor, or from `create_crosstalk_free_model`

**Comment by enielse** [comment]
I'm not sure if this should be closed yet, but several of the aforementioned items have been addressed in release 0.9.10.  In particular:
- the idle tomography issue should be fixed now, and more stringent unit tests have been added that should better catch when something goes wrong.
- the `"(auto_global_idle)"` gate should just be a reference to an idle gate (often of another name, e.g. `Gi`) constructed from the processor spec.  Noise should be able to be added to this idle gate like any other.  If this isn't the case, then we've got a bug to deal with.
